internal/bufferpool/bufferpool.go

```diff
@@ -23,24 +23,68 @@
 package bufferpool
 
 import (
-	"bytes"
+	"flag"
 	"sync"
 )
 
-var _bufferPool = sync.Pool{
-	New: func() interface{} {
-		return &bytes.Buffer{}
-	},
+var _pool = NewPool()
+
+// Option configures a buffer pool.
+type Option func(*Pool)
+
+// Pool represents a buffer pool with a set of options.
+type Pool struct {
+	testDetectUseAfterFree bool
+	pool                   sync.Pool
 }
 
-// Get returns a new Buffer from the Buffer pool that has been reset.
-func Get() *bytes.Buffer {
-	buf := _bufferPool.Get().(*bytes.Buffer)
-	buf.Reset()
+func init() {
+	// FIXME: This is a huge hack just for prototyping.
```

Comment:
any way around this?

Reply:
Unfortunately, I don't think there's any better way. I'll remove the FIXME and update the comment.